### PR TITLE
Improve Dashboards schema docs

### LINKED ISSUE #68479: Document Dashboard schema
## General guidance
For all the types, please specify:
1. what type of behavior do they drive in the UI (see https://github.com/grafana/grafonnet/blob/main/docs/grafonnet/dashboard.md#fn-withlivenow as an example); 
2. if the name of the field in the UI is different from the name of the type in the schema, and what the schema type refers to. If there is better information in the Grafana docs, please add a link;
3. If there are already specified examples in the Grafana Documentation, please include a link (see https://github.com/grafana/grafonnet/blob/main/docs/grafonnet/dashboard/variable.md#fn-querywithregex);
4. If there is a list of specific values, please explain what changes when using each of the values; 
5. Always specify the syntax of the value in the type, not just what it drives (example of how NOT to do it https://github.com/grafana/grafonnet/blob/main/docs/grafonnet/dashboard.md#fn-withtime); 

### Tier 1

- All the types that have specific values (like dropdowns) 
- Types where the value defines a specific behavior in the dashboard, like true/false behaviors or list behaviors (for example, the ones with a list of accepted values that drive UI changes but are not specifically a dropdown) As a reference, please take this example: https://github.com/grafana/grafonnet/blob/main/docs/grafonnet/dashboard/variable.md#fn-querywithsort

_[TODO: Add list of specific properties]_

### Tier 2

- All the types that have a mismatch with how the same is called in the UI. In this case, please follow the guidelines in the[ General guidance](https://docs.google.com/document/d/1HskiqX7iFswjEOIoHCVvnvtNevx4uNOO9E8P6CzuwFM/edit#heading=h.pd7c3w2ft573) section. 
- All types where the configuration can be complex and could be better explained with code examples, like https://github.com/grafana/grafonnet/blob/main/docs/grafonnet/dashboard/variable.md#fn-customnew

_[TODO: Add list of specific properties]_

### Tier 3
All the types that require a specific value that is a reference to something else inside Grafana, like references to OrgID, UserID etc. 
_[TODO: Add list of specific properties]_

### Tier 4
Free-hand types, types where the content is a description or text, numbers, strings that do not require a specific formatting.
_[TODO: Add list of specific properties]_

### PR DESCRIPTION
**Pull request by ivanortegaalba**
Focus on documenting the most tricky properties to have good [documentation for Grafonnet](https://github.com/grafana/grafonnet/blob/main/docs/grafonnet/dashboard.md#fn-withannotations).

Fixes: #68479

### CODE REVIEW

**Comment by malcolmholmes** [comment]
This is amazing. I'm hoping @Duologic will be able to review in more detail.

**Review by Duologic — APPROVED** [review]
From an as-code perspective, this looks very good, I love this!

File: packages/grafana-schema/src/raw/dashboard/x/dashboard_types.gen.ts [hunk 1]
```diff
@@ -517,17 +550,25 @@ export interface Panel {
...
    * TODO docs
    */
   timeRegions?: Array<unknown>;
   /**
-   * TODO docs
-   * TODO tighter constraint
+   * Overrides the time range for individual panels by shifting its start and end relative to the time picker.
+   * For example, you can shift the time range for the panel to be two hours earlier than the dashboard time picker setting `2h`.
+   * Note: Panel time overrides have no effect when the dashboard’s time range is absolute.
+   * See: https://grafana.com/docs/grafana/latest/panels-visualizations/query-transform-data/#query-options
```
Comment (ivanortegaalba) [inline packages/grafana-schema/src/raw/dashboard/x/dashboard_types.gen.ts:571]: @jdbaldry When we want to link to Grafana docs. What's the best way to do it? Thank you!

Comment (jdbaldry) [reply packages/grafana-schema/src/raw/dashboard/x/dashboard_types.gen.ts:571]: Hey Ivan, is this in the code itself? As far as I'm aware, there is no alternative to specifying the full URL. If it is in documentation, we would use a Hugo relref for build time link checking. More information about linking in documentation is in https://grafana.com/docs/writers-toolkit/writing-guide/references/.

**Pull request merged**